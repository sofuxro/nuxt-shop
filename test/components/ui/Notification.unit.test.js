import { factoryMount } from '~/test/utils/factories'
import flushPromises from 'flush-promises'

import Component from '~/components/ui/Notification'

let wrapper
const store = {
    state: {
        notificationMessage: ''
    },

    mutations: {
        addNotificationMessage(state, message) {
            state.notificationMessage = message
        },
        removeNotificationMessage: jest.fn((fn) => {
            store.state.notificationMessage = ''
        })
    }
}

beforeEach(() => {
    wrapper = factoryMount(Component, { store })
})

afterEach(() => {
    wrapper.destroy()
})

describe('Notification (a.k.a. snackbar - Vuetify term)', () => {
    test('initially is hidden', () => {
        expect(wrapper.find('.v-snack').exists()).toBe(false)
        // Final snapshot
        expect(wrapper.element).toMatchSnapshot()
    })

    test('closes when the button is clicked', async () => {
        wrapper.vm.$store.commit('addNotificationMessage', 'dummy message')
        await flushPromises()

        // Initial snapshot (when the notification is visible)
        expect(wrapper.element).toMatchSnapshot()

        // Clicking the close button
        wrapper.find('button').trigger('click')
        await flushPromises()
        expect(store.mutations.removeNotificationMessage).toBeCalled()

        expect(wrapper.find('.v-snack').exists()).toBe(false)
        // Final snapshot
        expect(wrapper.element).toMatchSnapshot()
    })
})
