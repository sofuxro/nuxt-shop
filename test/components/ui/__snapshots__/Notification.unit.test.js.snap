// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification (a.k.a. snackbar - Vuetify term) closes when the button is clicked 1`] = `<!---->`;

exports[`Notification (a.k.a. snackbar - Vuetify term) closes when the button is clicked 2`] = `
<div
  class="v-snack orange--text text--lighten-1 v-snack--active v-snack--bottom v-snack--right v-snack-transition-enter v-snack-transition-enter-active"
>
  <div
    class="v-snack__wrapper"
  >
    <div
      class="v-snack__content"
    >
      
    dummy message
    
      <button
        class="v-btn v-btn--flat v-btn--text theme--dark v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-close theme--dark orange--text text--lighten-1"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Notification (a.k.a. snackbar - Vuetify term) initially is hidden 1`] = `<!---->`;
